primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/csv/Csv002Test.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009-2023 PrimeTek Informatics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.integrationtests.csv;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.primefaces.component.selectoneradio.SelectOneRadio;
+import org.primefaces.integrationtests.colorpicker.AbstractColorPickerTest;
+import org.primefaces.selenium.AbstractPrimePage;
+import org.primefaces.selenium.component.CommandButton;
+import org.primefaces.selenium.component.Messages;
+import org.primefaces.selenium.component.SelectBooleanCheckbox;
+import org.primefaces.selenium.component.SelectManyCheckbox;
+import org.primefaces.selenium.component.Spinner;
+import org.primefaces.selenium.component.ToggleSwitch;
+import org.primefaces.selenium.component.TriStateCheckbox;
+
+public class Csv002Test extends AbstractColorPickerTest {
+
+    @Test
+    @Order(1)
+    @DisplayName("CSV: SelectBooleanCheckbox")
+    public void testSelectBooleanCheckbox(Page page) {
+        // Arrange
+
+        // Assert
+        Assertions.assertEquals("", page.msgSelectBooleanCheckbox.getText());
+
+        // Act
+        page.selectBooleanCheckbox.check();
+        page.selectBooleanCheckbox.uncheck();;
+
+        // Assert
+        Assertions.assertEquals("yes/no: Validation Error: Value is required.",
```

Comment:
Any idea what CSV should do with `p:selectBooleanCheckbox` and `p:triStateCheckbox` set to required?
![image](https://github.com/primefaces/primefaces/assets/10461942/79f2ba90-bf31-4199-b1a8-5b4373bff8dc)
IMO it does currently nothing.

(I´m adding CSV-IT´s as part of this PR too. This IT fails currently.)

Reply:
`@AssertTrue` helps (partial) for SelectBooleanCheckbox.